packages/publisher/electron-release-server/test/PublisherERS_spec.ts

```diff
@@ -0,0 +1,29 @@
+import { expect } from 'chai';
+import { ForgeConfig } from '@electron-forge/shared-types';
+import fetchMock from 'fetch-mock';
+import proxyquire from 'proxyquire';
+
+describe('PublisherERS', () => {
+  let fetch: typeof fetchMock;
+  beforeEach(() => {
+    fetch = fetchMock.sandbox();
+  });
+  it('fail if the server returns 4xx', async () => {
+    fetch.mock('begin:http://example.com', { body: {}, status: 400 });
+    const PublisherERS = proxyquire.noCallThru().load('../src/PublisherERS', {
+      'node-fetch': fetch,
+    }).default;
+
+
+    const publisher = new PublisherERS({
+      baseUrl: 'http://example.com',
+      username: 'test',
+      password: 'test',
+    });
+    return expect(publisher.publish({ makeResults: [], dir: '', forgeConfig: {} as ForgeConfig })).to.eventually.be.rejected;
```

Comment:
I'm pretty sure this test hits this error instead:

https://github.com/electron-userland/electron-forge/blob/fb725329682c06c527dc9aebef66581bfc25ff8a/packages/publisher/electron-release-server/src/PublisherERS.ts#L39-L41

The test should set `baseUrl`, `username`, and `password`. I suggest a test that has `to.eventually.be.rejectedWith(/^ERS publish failed with status code: 400/)`

Reply:
I made several test but this doesn't seem to work with regex, we have to put the full error message (I changed it this way)

`Publisher` is configured here (with `baseUrl`, etc): https://github.com/electron-userland/electron-forge/pull/1538/files#diff-32594fa1ac788919a60c060f6b27112eR18

Reply:
Ah, I missed that, sorry.